### ISSUE TITLE
Gesture Volume Slider is Glitching

### ISSUE THREAD

**Issue by SafwanLjd** [LagradOst/CloudStream-3#419]
When using the swiping gesture to change the volume level, the indicator bar glitches and flickers, as shown below:
![demonstration of the bug](https://user-images.githubusercontent.com/59485214/148710277-fecfe770-c8f2-4ed3-b5df-9955df36a1c4.gif)

**Comment by LagradOst** [comment]
unable to replicate, try latest preview

**Comment by SafwanLjd** [comment]
The issue is still there, my screen is 90Hz if that has anything to do with it…

I think that the issue might be that you're fetching the volume level on every update to the indicator bar… you can make this better by fetching it only once, when the user starts swiping, and adjust it according to the user input instead of fetching the volume level constantly, i.e.

* When the user swips X pixels, increment/decrement the indicator by X% and change the volume level alongside it

instead of:

* When the user swipes X pixels, increment/decrement the volume by X%, and then fetch the new volume level and make the bar represent it



I'm not sure that this is how it actually work, but this is what the behavior suggests, sorry for speculating, and thanks for your time

**Comment by LagradOst** [comment]
yes rn we fetch every frame, https://github.com/LagradOst/CloudStream-3/blob/0cddcb1864daa5b47aee0e0aac12a0a6431cbfd4/app/src/main/java/com/lagradost/cloudstream3/ui/player/FullScreenPlayer.kt#L767

**Comment by LagradOst** [comment]
the problem is that we have to at least get currentVolume when the volume is changed because otherwise it will break for older phones.

**Comment by SafwanLjd** [comment]
Maybe add it as an option in the settings?

There is probably a work around though… I mean how does applications like VLC do it?

**Comment by LagradOst** [comment]
making a setting will require more work then fixing it

**Comment by SafwanLjd** [comment]
I see… well, I hope you guys figure it out
